packages/java/endpoint/src/main/java/com/vaadin/hilla/route/ClientRouteRegistry.java

```diff
@@ -219,6 +219,24 @@ private void registerAndRecurseChildren(String basePath,
         }
     }
 
+    /**
+     * Checks if the route registry has a main layout.
+     * <ul>
+     * A main layout is an route that:
+     * </ul>
+     * <li>has an empty route path</li>
+     * <li>has a non-null children list (can be empty)</li>
+     * <li>has no parent</li>
+     *
+     * @return {@code true} if the registry has a main layout, {@code false}
+     *         otherwise
+     */
+    public synchronized boolean hasMainLayout() {
+        return getAllRoutes().values().stream()
```

Comment:
This would not work with a structure like below as route with a parent is not added in the `registeredRoutes` Map. it should probably check parent here too.

```
[{
 "route":""
 "title":"Layout",
 "children":[
	{
	 "route":""
	 "title":"Public"
	}
	]
}]
```

Reply:
Good catch! I changed the detection algorithm, and added tests for covering similar cases.